### PR TITLE
fix(backend): Fix timestamp in task store, add disabled unit test

### PR DESCRIPTION
**Pull request by gkcalat**
**Description of your changes:**

Previously, a unit test was removed due to discrepancy in how MySQL and SQLite handle insert/update queries. This PR add the unit test and simplifies the logic for assigning timestamps and returning a error message inside the `UpdateOrCreateTasks`.

/hold until #9245 gets merged.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/apiserver/storage/task_store_test.go [hunk 1]
```diff
@@ -515,3 +515,114 @@ func TestTaskStore_patchWithExistingTasks(t *testing.T) {
...
+		Fingerprint:       "10",
+		State:             model.RuntimeStateUnspecified,
+		StateHistory:      []*model.RuntimeStatus{{UpdateTimeInSec: 4, State: model.RuntimeStateUnspecified}},
+	}
+
+	tests := []struct {
+		name    string
+		tasks   []*model.Task
+		want    []*model.Task
+		wantErr bool
```
Comment (chensun) [inline backend/src/apiserver/storage/task_store_test.go:605]: nit: avoid dead code unless you have a negative test case.

**Review by chensun — COMMENTED** [review]
/lgtm

File: backend/src/apiserver/storage/task_store_test.go (hunk 1 above)
Comment (gkcalat) [reply backend/src/apiserver/storage/task_store_test.go]: Thanks! Updated.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**